### PR TITLE
Fix mac integration function tools path

### PR DESCRIPTION
**Pull request by VasuBhog**
If users on Mac have Azure Function core tools installed via [Brew](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Cmacos%2Ccsharp%2Cportal%2Cbash#install-the-azure-functions-core-tools). Then dotnet test will constantly fail.

### CODE REVIEW

File: test/Integration/IntegrationTestBase.cs [hunk 1]
```diff
@@ -265,6 +265,20 @@ private static string GetFunctionsCoreToolsPath()
...
+                    if (File.Exists(usrBinFuncPath))
+                    {
+                        return usrBinFuncPath;
+                    }
+                    throw new FileNotFoundException($"Azure Function Core Tools not found at {funcPath} or {usrBinFuncPath}");
+                }
+                else
+                {
+                    throw new FileNotFoundException($"Azure Function Core Tools not found at {funcPath}");
+                }
```
Comment (Charles-Gagnon) [inline test/Integration/IntegrationTestBase.cs:281]: This isn't needed, line 282 already throws if none of the if's work
```suggestion
```

File: test/Integration/IntegrationTestBase.cs [hunk 2]
```diff
@@ -265,6 +265,20 @@ private static string GetFunctionsCoreToolsPath()
                     }
                     throw new FileNotFoundException($"Azure Function Core Tools not found at {funcPath} or {programFilesFuncPath}");
                 }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (chlafreniere) [inline test/Integration/IntegrationTestBase.cs:268]: Does OSPlatform only has OSX for mac? Just making sure that it's inclusive for macos11/12/etc

Comment (VasuBhog) [reply test/Integration/IntegrationTestBase.cs:268]: There seems to be `FreeBSD, Linux, OSX, Windows, and Create OS` as options for OSPlatform. 

I believe that OSX should be inclusive of any version of Mac.

**Comment by chlafreniere** [comment]
@VasuBhog I tried pulling down the branch, had a connection string that used SQL Auth, and got an error about kerberos:

```
Cannot authenticate using Kerberos. Ensure Kerberos has been initialized on the client with 'kinit' and a Service Principal Name has been registered for the SQL Server to allow Kerberos authentication.
```

Is this a known issue?

**Comment by VasuBhog** [comment]
> ```
> Cannot authenticate using Kerberos. Ensure Kerberos has been initialized on the client with 'kinit' and a Service Principal Name has been registered for the SQL Server to allow Kerberos authentication.
> ```

Does not seem to be an issue that's tracked in the powershell worker repo. So please create an issue and then can follow up on investigating it.

**Comment by chlafreniere** [comment]
> > ```
> > Cannot authenticate using Kerberos. Ensure Kerberos has been initialized on the client with 'kinit' and a Service Principal Name has been registered for the SQL Server to allow Kerberos authentication.
> > ```
> 
> 
> Does not seem to be an issue that's tracked in the powershell worker repo. So please create an issue and then can follow up on investigating it.

Ah, user error, didn't have the SA_PASSWORD environment variable set.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**